dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -1133,6 +1134,34 @@ String DAGExpressionAnalyzer::appendNullAwareSemiJoinEqColumn(
     return and_arg_names.size() == 1 ? and_arg_names[0] : applyFunction("and", and_arg_names, last_step.actions, nullptr);
 }
 
+bool DAGExpressionAnalyzer::appendRuntimeFilterProperties(RuntimeFilterPtr runtime_filter) {
+    NameAndTypePair nameAndType;
+    if (isColumnExpr(runtime_filter->getSourceExpr())) {
+        nameAndType = getColumnNameAndTypeForColumnExpr(runtime_filter->getSourceExpr(), getCurrentInputColumns());
+        runtime_filter->setSourceColumnName(nameAndType.name);
+    } else {
+        return false;
+    }
+    Block header;
+    std::shared_ptr<Set> in_values_set;
```

Comment:
Why use shared_ptr?

Reply:
Because this value needs to be assigned to the property of RuntimeFilter. 
Runtime Filter itself needs to use shared ptr